### PR TITLE
Use cruft and Ouranosinc/cookiecutter-pypackage

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Changes

* Regenerated the project template using https://github.com/Ouranosinc/cookiecutter-pypackage so that changes can be tracked.
* Renamed `/doc` to `/docs`.
* Added a pre-commit configuration with `flake8` and `pydocstyle` checks disabled for now.
* Added Anaconda environment files for documentation and development.
* Cleaned up dependencies and commented out all broken cells in `getting-started.ipynb`.
* Updated `pre-commit` hook versions.
* `xdatasets` now uses `pyproject.toml` with `flit` build backend

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@sebastienlanglois 

I've cleaned this up and will be making a squash merge to not pollute the git history.

It looks like there's an issue with one of the notebook cells, but otherwise the package is in a more maintainable condition, IMO. Feel free to add changes to the notebook here. Let me know if there's anything you'd like changed.

Best,

**Comment by Zeitsperre** [comment]
@sebastienlanglois

This PR is good to merge. Care to give it a final review? Let me know!

**Review by sebastienlanglois — APPROVED** [review]

**Comment by sebastienlanglois** [comment]
@Zeitsperre 
Thanks for all the work and sorry for the delay. I'm ready to merge this branch but noticed you were still adding some commits recently. Should I wait for more changes or can I merge it now ? Thanks!

File: xdatasets/core.py [hunk 1]
```diff
@@ -1,87 +1,82 @@
...
+
+    The Query interface facilitates access to analysis-ready earth observation datasets and allows for
+    spatiotemporal operations to be performed based on user queries.
 
-    """"The Query interface facilitates access to analysis-ready 
-    earth observation datasets and allows for spatiotemporal
-    operations to be performed based on user queries.
-        
-    Parameters
+    Attributes
```
Comment (RondeauG) [inline xdatasets/core.py:30]: Why is it getting renamed `Attributes`?

File: xdatasets/core.py [hunk 2]
```diff
@@ -1,87 +1,82 @@
...
-              ===========  ==============  
-              variables    str, List[str]  
-              ===========  ============== 
-        
-        The list of available datasets in this library can be accessed here:
-        # Coming soon!
+        If a str, a dataset name, i.e.: era5_land_reanalysis.
+        If a list, a list of dataset names, i.e.: [era5_single_levels_reanalysis, era5_land_reanalysis].
+        If a dictionary, it should map dataset names to their corresponding requested
+        content such as some desired variables. The list of available datasets in this library is coming soon!
```
Comment (RondeauG) [inline xdatasets/core.py:36]:
```suggestion
        content such as some desired variables. See the notes below for more details. The list of available datasets in this library is coming soon!
```

File: xdatasets/core.py [hunk 3]
```diff
@@ -237,34 +230,41 @@ def load_query(self,
         # Load data for each dataset
         dsets = []
         for dataset_name in datasets_name:
-            data = None
+            # data = None
```
Comment (RondeauG) [inline xdatasets/core.py:233]:
```suggestion
            # data = None  # FIXME: This is never used.
```

File: xdatasets/spatial.py [hunk 4]
```diff
@@ -1,110 +1,128 @@
...
-import pandas as pd
 
 from .utils import HiddenPrints
 
+try:
+    import xagg as xa
+except ImportError:
+    warnings.warn("xagg is not installed. Please install it with `pip install xagg`")
+    xa = None
+
```
Comment (RondeauG) [inline xdatasets/spatial.py:16]: @sebastienlanglois Are you OK with that and the change below that raises an exception if someone tries to use `create_weights_mask` without `xagg`?

File: xdatasets/spatial.py [hunk 5]
```diff
@@ -1,110 +1,128 @@
...
 
 
 def clip_by_point(ds, space, dataset_name):
-
-    #TODO : adapt logic for coordinate names
+    # TODO : adapt logic for coordinate names
 
     logger = logging.getLogger()
-    logging.info(f"Spatial operations: processing points with {dataset_name}")
+    logger.info(f"Spatial operations: processing points with {dataset_name}")
```
Comment (RondeauG) [inline xdatasets/spatial.py:119]: `logger` is never set elsewhere, for example in `clip_by_bbox`. Should we instead git it as it was, but remove `logger`?

File: xdatasets/spatial.py (hunk 4 above)
Comment (sebastienlanglois) [reply xdatasets/spatial.py:16]: Yes I'm good with that

File: xdatasets/__init__.py [hunk 6]
```diff
@@ -1,6 +1,8 @@
...
+"""Easy access to Earth observation datasets with xarray."""
+
 from .core import Query
+from .tutorial import list_available_datasets, load_dataset
 
 __author__ = """Sebastien Langlois"""
 __email__ = "sebastien.langlois62@gmail.com"
-__version__ = '0.2.10'
\ No newline at end of file
+__version__ = "0.3.0"
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply xdatasets/__init__.py:8]: I think so. This is a pretty significant departure, IMO.

File: xdatasets/core.py (hunk 1 above)
Comment (Zeitsperre) [reply xdatasets/core.py:30]: It has to do with the numpy-docstring standard: `Parameters` are unique to functions, while classes have `Attributes`. It also affects the rendering on the documentation.

Further reading: https://www.datasciencebyexample.com/2023/05/25/python-class-paramter-vs-instance-attribute/

File: xdatasets/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (RondeauG) [reply xdatasets/__init__.py:8]: OK, I was just making sure!

File: xdatasets/spatial.py (hunk 5 above)
Comment (Zeitsperre) [reply xdatasets/spatial.py:119]: If we aren't going to employ conventional logging practices (setting loggers by file/scope), we're probably safer not instantiating `logger` at all here.

File: .readthedocs.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# .readthedocs.yml
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply .readthedocs.yml:1]: We can leave the file as is and set things up on RTD later. In all honesty, I wanted to enable both `xdatasets` and `xhydro` on RTD once this is in.

**Comment by Zeitsperre** [comment]
@sebastienlanglois

Good to hear from you! This PR is good. For the documentation, I'm going to open a new issue about it and do the final touches there. There's enough going on in here.

Feel free to merge!

File: .github/workflows/gh-pages.yml [hunk 8]
```diff
@@ -1,36 +1,46 @@
 name: Docs
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply .github/workflows/gh-pages.yml:1]: These will be very simple to address in another PR. I'll be disabling this workflow and enabling RTD at the same time.

**Pull request merged**